rust_src/src/vectors.rs

```diff
@@ -139,25 +139,28 @@ impl LispVectorRef {
     pub fn as_slice(&self) -> &[LispObject] {
         unsafe {
             slice::from_raw_parts(
-                mem::transmute::<_, *const LispObject>(&self.contents),
+                &self.contents as *const [i64; 1] as *const LispObject,
+                //mem::transmute::<_, *const LispObject>(&self.contents),
                 self.len(),
             )
         }
     }
 
     #[inline]
-    pub fn as_mut_slice(&self) -> &mut [LispObject] {
+    pub fn as_mut_slice(&mut self) -> &mut [LispObject] {
         unsafe {
             slice::from_raw_parts_mut(
-                mem::transmute::<_, *mut LispObject>(&self.contents),
+                &self.contents as *const [i64; 1] as *mut LispObject,
```

Comment:
Isn't that `i64` platform dependent?

Reply:
Why? Lisp_Object is a 64bit int.

Reply:
Even on 32bit platforms?

Reply:
I had thought so. But it is a `c_long`. Either way, a naked i64 is wrong there.